asab/web/authz/decorator.py

```diff
@@ -0,0 +1,56 @@
+import functools
+
+import aiohttp.web
+import aiohttp.hdrs
+
+from ...config import Config
+
+
+def required(*resources):
+	'''
+	Checks that user authorized with access token in
+	Authorization header has access to a given tenant space
+	using SeaCat Auth RBAC authorization.
+
+	Example of use:
+
+	@asab.web.tenant.tenant_handler
+	@asab.web.authz.required("tenant:access")
+	async def endpoint(self, request, *, tenant):
+		...
+	'''
+
+	def decorator_required(func):
+
+		@functools.wraps(func)
+		async def wrapper(*args, **kargs):
+			request = args[-1]
+
+			# Check if tenant exists in the request
+			if not hasattr(request, "Tenant"):
+				raise aiohttp.web.HTTPUnauthorized()
+
+			tenant = request.Tenant
+			rbac_url = Config["authz"]["rbac_url"]
+
+			# Check authorization using RBAC
+			# Authorization header should already be part of the request
+			for resource in resources:
```

Comment:
This has to be cached in some reasonable way.
Ideally using some kind of credentials session indicator in the HTTP request.

But for a starter, ~5 minute cache and TODO will be ok